torch/distributed/algorithms/ddp_comm_hooks/default_hooks.py

```diff
@@ -65,3 +67,78 @@ def decompress(fut):
         return [decompressed_tensor]
 
     return fut.then(decompress)
+
+
+class AllReduceFirstKStepsState(object):
+    """
+    State information required by allreduce_first_k_steps_hook to run gradient averaging for the first K steps.
+    Need to specify ``k``.
+    """
+
+    __slots__ = [
+        "process_group",
+        "k",
+        "step",
+    ]
+
+    def __init__(
+        self,
+        process_group,
+        k,
+    ):
+        self.process_group = process_group
+        self.k = k
+        self.step = 0
+
+        logging.info(
+            "Allreduce-first-k-steps hook runs allreduce for the first {} steps.".format(
+                self.k,
+            )
+        )
+
+    def maybe_increase_step(self, bucket):
```

Comment:
nit: I remember we have a similar function for other hooks that keep track of iteration. Could we consolidate that logic, maybe by a simple counter function/class?

Reply:
They are similar, applied to different state classes. The main difference is the log info and the variable name that represents the last iteration.

To consolidate them, need to expose a helper function as a new public API. However, this hook is very experimental, as it is still unclear to support post-local SGD with DDP, prefer no public API changes based on this hook at this time.

Added a TODO here.